cmake/clang_flags.cmake

```diff
@@ -0,0 +1,87 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# If we are running clang >= 10.0 then enable more checking. Some of these warnings may not exist
+# in older versions of clang so we limit the use of older clang for these checks.
+if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
```

Comment:
Move to cmake/modules/ClangWarnings.cmake, we only keep config.cmake under the cmake folder.